### PR TITLE
chore: Add eslint-plugin-eslint-plugin to warn about potential issues

### PR DESCRIPTION
**Pull request by armano2**
I did set up most common rules from it, but it will be nice if we will have also:
- require-meta-docs-url
- require-meta-type
- report-message-format

see: https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin

### CODE REVIEW

File: tests/lib/rules/type-annotation-spacing.js [hunk 1]
```diff
@@ -4234,11 +4229,6 @@ type Bar = Record<keyof Foo, string>
             ],
             parser: "typescript-eslint-parser",
         },
-        `
-interface resolve {
-    resolver?: (() => PromiseLike<T>) | PromiseLike<T>;
-}
-        `,
```
Comment (j-f1) [inline tests/lib/rules/type-annotation-spacing.js]: Why the removed tests?

Comment (armano2) [reply tests/lib/rules/type-annotation-spacing.js]: this test is duplicated 3 times

Comment (armano2) [reply tests/lib/rules/type-annotation-spacing.js]: https://github.com/bradzacher/eslint-plugin-typescript/blob/master/tests/lib/rules/type-annotation-spacing.js#L25
https://github.com/bradzacher/eslint-plugin-typescript/blob/master/tests/lib/rules/type-annotation-spacing.js#L3458
https://github.com/bradzacher/eslint-plugin-typescript/blob/master/tests/lib/rules/type-annotation-spacing.js#L4237

**Comment by bradzacher** [comment]
for `require-meta-docs-url`, we should probably wrap it up in a utility function so we don't have to worry about having the raw URL in every single rule.

Agree with `report-message-format`. 

Agree with `require-meta-type`. But we can turn that on in a separate PR (considering it'll need a change to every single rule).


Would also like (later, as they'll cause too many errors for this PR):
- `prefer-output-null`
- `no-deprecated-context-methods`
- `test-case-shorthand-strings`

I'll raise issues for the later ones to track it

**Comment by armano2** [comment]
that's why i did not enable them :)

**Comment by bradzacher** [comment]
ah. all good then I've raised the issues.

- prefer-output-null #212
- no-deprecated-context-methods #213
- test-case-shorthand-strings #214
- report-message-format #215
- require-meta-type #216
- require-meta-docs-url #217

**Pull request merged**